ERCS/erc-7540.md

```diff
@@ -224,30 +313,103 @@ MUST be emitted when a redemption Request is submitted using the `requestRedeem`
+
+This function MUST return `TODO` upon successful execution of the callback.
+
+```yaml
+- name: onERC7540DepositReceived
+  type: function
+
+  inputs:
+    - name: operator
+      type: address
+    - name: owner
+      type: address
+    - name: requestId
+      type: uint256
+    - name: data
+      type: bytes
+  outputs:
+    - name: interfaceId
+      type: bytes4
+```
+
+
+#### ERC7540RedeemReceiver
+The interface to be called on `requestRedeem`.
+
+`operator` is the `msg.sender` of the original `requestRedeem` call. `owner` is the `owner` of the `requestRedeem`. `requestId` is the output `requestId` of the `requestRedeem` and `data` is the `data` of the `requestRedeem`.
+
+This function MUST return `TODO` upon successful execution of the callback.
+
+```yaml
+- name: onERC7540RedeemReceived
+  type: function
+
+  inputs:
+    - name: operator
+      type: address
+    - name: owner
+      type: address
+    - name: requestId
+      type: uint256
+    - name: data
+      type: bytes
+  outputs:
+    - name: interfaceId
+      type: bytes4
+```
+
 ### [ERC-165](./eip-165.md) support
 
-Smart contracts implementing this standard MUST implement the [ERC-165](./eip-165.md) `supportsInterface` function.
+Smart contracts implementing this Vault standard MUST implement the [ERC-165](./eip-165.md) `supportsInterface` function.
 
-Asynchronous deposit Vaults MUST return the constant value `true` if `0xea446681` is passed through the `interfaceID` argument.
+Asynchronous deposit Vaults MUST return the constant value `true` if `0x1683f250` is passed through the `interfaceID` argument.
 
-Asynchronous redemption Vaults MUST return the constant value `true` if `0x2e9dd5bd` is passed through the `interfaceID` argument.
+Asynchronous redemption Vaults MUST return the constant value `true` if `0x0899cb0b` is passed through the `interfaceID` argument.
 
 ## Rationale
 
+### Including Request IDs but Not Including a Claim by ID method
+Requests in an Asynchronous Vault have properties of NFTs or Semi-Fungible tokens due to their asynchronicity. However, trying to pigeonhole all ERC-7540 Vaults into supporting [ERC-721](./eip-721) or [ERC-1155](./erc-1155) for Requests would create too much interface bloat. 
+
+Using both an id and address to discriminate Requests allows for any of these use cases to be developed at an external layer without adding too much complexity to the core interface.
+
+Certain Vaults especially `requestId==0` cases benefit from using the underlying ERC4626 methods for claiming because there is no discrimination at the `requestId` level. This standard is written primarily with those use cases in mind. A future standard can optimize for nonzero RID with support for claiming and transferring requests discriminated also with an `requestId`.
```

Comment:
```suggestion
Certain Vaults especially `requestId==0` cases benefit from using the underlying [ERC-4626](./eip-4626) methods for claiming because there is no discrimination at the `requestId` level. This standard is written primarily with those use cases in mind. A future standard can optimize for nonzero request ID with support for claiming and transferring requests discriminated also with an `requestId`.
```